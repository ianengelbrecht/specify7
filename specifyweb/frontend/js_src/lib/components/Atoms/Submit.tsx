// Force passing children by nesting rather than through the [value] attribute
import type { LocalizedString } from 'typesafe-i18n';

import { className } from './className';
import type { TagProps } from './wrapper';
import { wrap } from './wrapper';

type SubmitProps = {
  readonly children: LocalizedString;
  readonly value?: undefined;
  readonly title?: LocalizedString | undefined;
  readonly 'aria-label'?: LocalizedString | undefined;
};
const submitButton = (name: string, buttonClassName: string) =>
  wrap<'input', SubmitProps>(
    name,
    'input',
    buttonClassName,
    ({
      children,
      ...props
    }: SubmitProps & TagProps<'input'>): TagProps<'input'> => ({
      type: 'submit',
      ...props,
      value: children,
    })
  );
export const Submit = {
  Small: submitButton(
    'Submit.Small',
    `${className.smallButton} ${className.defaultSmallButtonVariant}`
  ),
  Fancy: submitButton(
    'Submit.Fancy',
    `${className.niceButton} ${className.fancyButton} !inline`
  ),
  Gray: submitButton(
    'Submit.Gray',
    `${className.niceButton} ${className.grayButton}`
  ),
  Red: submitButton(
    'Submit.Red',
    `${className.niceButton} ${className.redButton}`
  ),
  Blue: submitButton(
    'Submit.Blue',
    `${className.niceButton} ${className.blueButton}`
  ),
  Orange: submitButton(
    'Submit.Orange',
    `${className.niceButton} ${className.orangeButton}`
  ),
  Green: submitButton(
    'Submit.Green',
    `${className.niceButton} ${className.greenButton}`
  ),
} as const;
