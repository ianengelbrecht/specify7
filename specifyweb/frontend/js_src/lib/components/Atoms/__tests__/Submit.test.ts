import { snapshot } from '../../../tests/reactUtils';
import { Submit } from '../Submit';

snapshot(Submit.Small, { children: 'Submit' });
snapshot(Submit.Fancy, { children: 'Submit' });
snapshot(Submit.Gray, { children: 'Submit' });
snapshot(Submit.Red, { children: 'Submit' });
snapshot(Submit.Blue, { children: 'Submit' });
snapshot(Submit.Orange, { children: 'Submit' });
snapshot(Submit.Green, { children: 'Submit' });
