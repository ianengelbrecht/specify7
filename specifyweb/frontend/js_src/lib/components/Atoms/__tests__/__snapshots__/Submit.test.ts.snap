// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit.Blue renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm dialog-icon-info hover:bg-blue-700 bg-blue-600 text-white"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;

exports[`Submit.Fancy renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm bg-gray-300 hover:bg-brand-200 dark:bg-neutral-600
    hover:dark:bg-brand:400 text-gray-800 dark:text-white text-center !inline"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;

exports[`Submit.Gray renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm button hover:bg-gray-400 bg-gray-300 text-gray-800 shadow-sm
    dark:bg-neutral-600 dark:text-gray-100 hover:dark:bg-neutral-500"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;

exports[`Submit.Green renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm dialog-icon-success hover:bg-green-800 bg-green-700 text-white"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;

exports[`Submit.Orange renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm dialog-icon-warning hover:bg-orange-600 bg-orange-500 text-white"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;

exports[`Submit.Red renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm dialog-icon-error hover:bg-red-800 bg-red-700 text-white"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;

exports[`Submit.Small renders without errors 1`] = `
<DocumentFragment>
  <input
    class="button rounded cursor-pointer active:brightness-80 px-4 py-2
    disabled:bg-gray-200 disabled:text-gray-500 dark:disabled:!bg-neutral-700 gap-2
    inline-flex items-center capitalize justify-center shadow-sm !py-1 !px-2 button hover:bg-gray-400 bg-gray-300 text-gray-800 shadow-sm
    dark:bg-neutral-600 dark:text-gray-100 hover:dark:bg-neutral-500 ring-1 ring-gray-400 shadow-sm dark:ring-0
    disabled:ring-gray-400 disabled:dark:ring-neutral-500 hover:bg-brand-200 dark:hover:bg-brand-400"
    type="submit"
    value="Submit"
  />
</DocumentFragment>
`;
